File: programacao-em-python/Exercicios/banco/ContaPoupanca.py
from ContaBancaria import ContaBancaria

class ContaPoupanca(ContaBancaria):
    def __init__(self, numero, saldo, cliente):
        super().__init__(numero, saldo, cliente)
        self._taxa_juros = 0.005

        def get_taxa_juros(self):
            return self._taxa_juros
        
        def set_taxa_juros(self, taxa_juros):
            self._taxa_juros = taxa_juros

        def aplicar_juros(self):
            self._saldo += self._saldo * self._taxa_juros

File: ProjetosPOO/ContasBancarias/src/principal/ContaPoupanca.java
package principal;

/**
 *
 * @author wescleyj
 */

public class ContaPoupanca extends ContaBancaria{
    private double taxaRendimento;
    
    public ContaPoupanca() {
        taxaRendimento = 0;
    }

    public ContaPoupanca(double taxaRendimento, String numeroConta, String titular, double saldo) {
        super(numeroConta, titular, saldo);
        this.taxaRendimento = taxaRendimento / 100;
    }
    
    
    
    public void aplicarRendimento() {
        setSaldo(getSaldo() + (taxaRendimento * getSaldo()));
    }
    
    @Override
    public String toString() {
        return "Conta Poupança {" +
               "Numero da conta = '" + getNumeroConta() + '\'' +
               " Titular da conta = '" + getTitular() + '\'' +
               " Saldo da conta = '" + getSaldo() + '\'' +
               " Taxa de rendimento = '" + taxaRendimento * 100 + "%" + '\'' + 
               '}';
    }

    public double getTaxaRendimento() {
        return taxaRendimento;
    }

    public void setTaxaRendimento(double taxaRendimento) {
        this.taxaRendimento = taxaRendimento;
    }
    
    
}
